cmd/panacead/cmd/init.go

```diff
@@ -0,0 +1,142 @@
+	}
+
+	var stakingGenState stakingtypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[stakingtypes.ModuleName], &stakingGenState); err != nil {
+		return nil, err
+	}
+	stakingGenState.Params.UnbondingTime = time.Second * 60 * 60 * 24 * 7 * 3 // three weeks
+	stakingGenState.Params.MaxValidators = 50
+	stakingGenState.Params.BondDenom = assets.MicroMedDenom
+	appState[stakingtypes.ModuleName] = cdc.MustMarshalJSON(&stakingGenState)
+
+	var mintGenState minttypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[minttypes.ModuleName], &mintGenState); err != nil {
+		return nil, err
+	}
+	mintGenState.Minter = minttypes.InitialMinter(sdk.NewDecWithPrec(7, 2)) // 7% inflation
+	mintGenState.Params.MintDenom = assets.MicroMedDenom
+	mintGenState.Params.InflationRateChange = sdk.NewDecWithPrec(3, 2) // 3%
+	mintGenState.Params.InflationMin = sdk.NewDecWithPrec(7, 2)        // 7%
+	mintGenState.Params.InflationMax = sdk.NewDecWithPrec(10, 2)       // 10%
+	mintGenState.Params.BlocksPerYear = uint64(60 * 60 * 24 * 365.25)  // assuming 1 second block time
+	appState[minttypes.ModuleName] = cdc.MustMarshalJSON(&mintGenState)
+
+	var distrGenState distrtypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[distrtypes.ModuleName], &distrGenState); err != nil {
+		return nil, err
+	}
+	distrGenState.Params.CommunityTax = sdk.ZeroDec()
+	appState[distrtypes.ModuleName] = cdc.MustMarshalJSON(&distrGenState)
+
+	var govGenState govtypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[govtypes.ModuleName], &govGenState); err != nil {
+		return nil, err
+	}
+	minDepositTokens := sdk.TokensFromConsensusPower(100000) // 100,000 MED
+	govGenState.DepositParams.MinDeposit = sdk.Coins{sdk.NewCoin(assets.MicroMedDenom, minDepositTokens)}
+	govGenState.DepositParams.MaxDepositPeriod = 60 * 60 * 24 * 14 * time.Second // 14 days
+	govGenState.VotingParams.VotingPeriod = 60 * 60 * 24 * 14 * time.Second      // 14 days
+	appState[govtypes.ModuleName] = cdc.MustMarshalJSON(&govGenState)
+
+	var crisisGenState crisistypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[crisistypes.ModuleName], &crisisGenState); err != nil {
+		return nil, err
+	}
+	crisisGenState.ConstantFee = sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(1000000000000)) // Spend 1,000,000 MED for invariants check
+	appState[crisistypes.ModuleName] = cdc.MustMarshalJSON(&crisisGenState)
+
+	var slashingGenState slashingtypes.GenesisState
+	if err := cdc.UnmarshalJSON(appState[slashingtypes.ModuleName], &slashingGenState); err != nil {
+		return nil, err
+	}
+	slashingGenState.Params.SignedBlocksWindow = 10000
+	slashingGenState.Params.MinSignedPerWindow = sdk.NewDecWithPrec(5, 2)
+	slashingGenState.Params.SlashFractionDoubleSign = sdk.NewDecWithPrec(5, 2) // 5%
+	slashingGenState.Params.SlashFractionDowntime = sdk.NewDecWithPrec(1, 4)   // 0.01%
+	appState[slashingtypes.ModuleName] = cdc.MustMarshalJSON(&slashingGenState)
```

Comment:
@youngjoon-lee 
I think the previous setting was a three-week period.
Is it the same setting?
https://github.com/medibloc/panacea-core/blob/985271f795c2ff0bf1f6b63152d97fabb41dbfa2/init/init.go#L125

Reply:
Oh, good point! I guess the `MaxEvidenceAge` parameter has moved from `x/slashing` to a new module `x/evidence`: https://github.com/cosmos/cosmos-sdk/blob/master/CHANGELOG.md#state-machine-breaking-3.
![image](https://user-images.githubusercontent.com/5462944/120127509-81499500-c1fa-11eb-9188-1f22290e3d12.png)
I forgot to manage it. Let me fix it.

Reply:
e18e8e6
Please review this additional commit.

As a background,
`MaxEvidenceAge` moved from `x/slahsing` to `x/evidence` at [Cosmos SDK v0.38.0](https://github.com/cosmos/cosmos-sdk/blob/master/CHANGELOG.md#state-machine-breaking-3) as mentioned above. Then, it moved to Tendmermint consensus params at [Cosmos SDK v0.40.0](https://github.com/cosmos/cosmos-sdk/blob/master/CHANGELOG.md#state-machine-breaking-2) as below:
![image](https://user-images.githubusercontent.com/5462944/120158008-9a6c3900-c22e-11eb-8da9-4b97cdb417ab.png)

As a result, the `genesis.json` should look like:
```json
{
  "consensus_params": {                                                                                                                                                                                                                                     
    "evidence": {                                                                                                                       
      "max_age_num_blocks": "1814400",                                                                                                  
      "max_age_duration": "1814400000000000",
      ...                                                                                           
    },
    ...                                                                                     
  }, 
  "app_state": {
    "slashing": {
      "params": {
         "signed_blocks_window": "10000",                                                                                                
         "min_signed_per_window": "0.050000000000000000",                                                                                
         "downtime_jail_duration": "600s",                                                                                               
         "slash_fraction_double_sign": "0.050000000000000000",                                                                           
         "slash_fraction_downtime": "0.000100000000000000"
      }
    },
    ...
  }
}
```

For more details about Tendermint consensus params, see https://docs.tendermint.com/master/tendermint-core/using-tendermint.html#fields.